server.go

```diff
@@ -306,6 +306,55 @@ func (s *server) AddFundStatus(ctx context.Context, in *breez.AddFundStatusReque
 	return &breez.AddFundStatusReply{Statuses: statuses}, nil
 }
 
+func (s *server) GetPayment(ctx context.Context, in *breez.GetPaymentRequest) (*breez.GetPaymentReply, error) {
+	redisConn := redisPool.Get()
+	m, err := redis.StringMap(redisConn.Do("HGETALL", "input-address:"+in.Address))
+	if err != nil {
+		log.Println("GetPayment error:", err)
+		return nil, status.Errorf(codes.Internal, "failed to retrieve address information")
+	}
+
+	//ensure we didn't pay this invoice before
+	preImage := m["payment:PaymentPreimage"]
+	if len(preImage) > 0 {
+		return nil, status.Errorf(codes.Internal, "payment already sent")
+	}
+
+	//ensure we have a payment request and the transactino amount
+	paymentRequest := m["paymentRequest"]
+	if len(paymentRequest) == 0 {
+		return nil, status.Errorf(codes.Internal, "payment request not found")
+	}
+	amt, err := strconv.ParseInt(m["tx:Amount"], 10, 64)
+	if err != nil || amt == 0 {
+		return nil, status.Errorf(codes.Internal, "failed to retrieve address information")
```

Comment:
Nit: Failed to parse amount. Also, perhaps verify that the amount is positive..

Reply:
fixed